bigframes/core/tools/datetimes.py

```diff
@@ -66,6 +61,45 @@ def to_datetime(
             f"Unit parameter is not supported for non-numerical input types. {constants.FEEDBACK_LINK}"
         )
 
+    if not utc and arg.dtype in ("string", "string[pyarrow]"):
+        if format:
+            raise NotImplementedError(
+                f"Customized formats are not supported for string inputs when utc=False. Please set utc=True if possible. {constants.FEEDBACK_LINK}"
+            )
+
+        assert not utc
+        assert format is None
+        assert unit is None
+        result = arg._apply_unary_op(  # type: ignore
+            ops.ToDatetimeOp(
+                utc=utc,
+                format=format,
+                unit=unit,
+            )
+        )
+        # Cast to DATETIME shall succeed if all inputs are tz-naive.
+        if not result.isnull().any():
+            return result
```

Comment:
We are using a safe_cast here to test out if all inputs are timezone-naive. Null output indicates that the time strings are timezone-aware.

Reply:
It seems that a null input becomes NaT in pandas, rather than failing the whole operation. So I think None->None is fine, I think the problem is when you have String->None indicating the cast from string to date failed.

Reply:
Good catch! And thank you for the fix!!